packages/react-components/src/trace-explorer/trace-explorer-views-widget.tsx

```diff
@@ -181,4 +169,67 @@ export class ReactAvailableViewsWidget extends React.Component<ReactAvailableVie
     private onOrderChange(ids: number[]): void {
         this.setState({ orderedNodes: ids });
     }
+
+    private listToTree(list: OutputDescriptor[]): TreeNode[] {
+        const rootNodes: TreeNode[] = [];
+        const lookup: { [key: string]: TreeNode } = {};
+        const idStringToNodeId: { [key: string]: number } = {};
+
+        // Fill-in the lookup table
+        list.forEach(output => {
+            const node: TreeNode = this.entryToTreeNode(output, idStringToNodeId);
+            lookup[output.id] = node;
+            this._nodeIdToOutput[node.id] = output;
+        });
+        // Create the tree in the order it has been received
+        list.forEach(output => {
+            const node = lookup[output.id];
+            if (output.parentId !== undefined) {
+                const parent: TreeNode = lookup[output.parentId];
+                if (parent) {
+                    if (parent.id !== node.id) {
```

Comment:
This protects against the model indicating parentId pointing to the node itself?

Reply:
yes